neurodsp/sim/periodic.py

```diff
@@ -179,6 +179,97 @@ def sim_bursty_oscillation(n_seconds, fs, freq, burst_def='prob', burst_params=N
     return sig
 
 
+@normalize
+def sim_variable_oscillation(fs, freqs, n_seconds='auto', cycle='sine', phase=0, **cycle_params):
+    """Simulate an oscillation that varies in frequency and cycle parameters.
+
+    Parameters
+    ----------
+    fs : float
+        Signal sampling rate, in Hz.
+    freqs : float or list
+        Oscillation frequencies.
+    n_seconds : 'auto' or float, optional, default: 'auto'
```

Comment:
I think I would vote for using 'None', instead of the string 'auto' here, for the in which `n_seconds` is to be inferred from frequency parameters. I think this would be consistent with other code (if value is not given, infer it from other params). This stops the param being mixed type - and I think should general work with the code (?).